Code/Framework/AzToolsFramework/AzToolsFramework/UI/Outliner/EntityOutlinerListModel.cpp

```diff
@@ -117,8 +117,12 @@ namespace AzToolsFramework
         ContainerEntityNotificationBus::Handler::BusConnect(editorEntityContextId);
 
         m_editorEntityUiInterface = AZ::Interface<AzToolsFramework::EditorEntityUiInterface>::Get();
-        AZ_Assert(m_editorEntityUiInterface != nullptr,
-            "EntityOutlinerListModel requires a EditorEntityUiInterface instance on Initialize.");
+        AZ_Assert(m_editorEntityUiInterface != nullptr, "EntityOutlinerListModel requires a EditorEntityUiInterface instance on Initialize.");
+
+        m_readOnlyEntityPublicInterface = AZ::Interface<AzToolsFramework::ReadOnlyEntityPublicInterface>::Get();
```

Comment:
where is the AzToolsFramework::ReadOnlyEntityPublicInterface being created?

Reply:
The interface is handled by the ReadOnlyEntitySystemComponent
https://github.com/o3de/o3de/pull/6059/files#diff-a262f9460deb797b6dc693d67b7c3ea678bdf9b24355c29e048a421bdf6a7162